types/pouchdb-mapreduce/index.d.ts

```diff
@@ -94,17 +94,32 @@ declare namespace PouchDB {
          * Invoke a map/reduce function, which allows you to perform more complex queries
          * on PouchDB than what you get with allDocs().
          */
-        query<Model>(fun: string | Filter | ((doc: any) => void), opts: Query.Options, callback: (err: Core.Error, res: Query.Response<Content & Model>) => void): void;
+        query<Model>(fun: string | ((doc: Content & Model) => void), opts: Query.Options, callback: (err: Core.Error, res: Query.Response<Content & Model>) => void): void;
```

Comment:
Not using the Callback type was where I found this. Easy enough to switch to that.

Looking at PouchDB docs it looks like Filter object without a Reduce is invalid and the reduce function may actually need to be Couch format: `(key: Array<[string, string]> | null, values: Content[] | Reduction[], rereduce: boolean) => Reduction`.